### ISSUE TITLE
Batch effect controlled in FindAllMarkers after IntegrateData?

### ISSUE THREAD

**Issue by malisas** [zhangzlab/covid_balf#2]
Hi, nice paper and thanks for sharing your code and data publicly!

I have a question about controlling for batch effect.

In `seurat_integration.R`, after you run `IntegrateData()`, you perform differential expression analysis using the original (uncorrected) expression values stored in the `"RNA"` assay. This choice makes sense to me. 

You [then run MAST to find markers for each cluster](https://github.com/zhangzlab/covid_balf/blob/080b84697409dc2705e55fb2850b958805160721/seurat_integration.R#L191):

```{r}
DefaultAssay(nCoV.integrated) <- "RNA"
# find markers for every cluster compared to all remaining cells, report only the positive ones
nCoV.integrated@misc$markers <- FindAllMarkers(object = nCoV.integrated, assay = 'RNA',only.pos = TRUE, test.use = 'MAST')
```

Since you are using the original values, I am thinking that it is important that the DE analysis controls for batch effect. Does the above code account for batch effect?

I was under the impression that the Seurat pipeline does not currently offer a way to include batch effect as a covariate, so I am very curious how people are handling this.

Apologies if I am overlooking an important point, as I am new to the Seurat pipeline. 
Thanks!

**Comment by zhangzlab** [comment]
Yes, this is a common question for Seurat. As you can see in the Seurat FAQ, using raw count is recommended (https://satijalab.org/seurat/faq). 
> We recommend running your differential expression tests on the “unintegrated” data. By default this is stored in the “RNA” Assay. There are several reasons for this. (1)The integration procedure inherently introduces dependencies between data points. This violates the assumptions of the statistical tests used for differential expression. (2)DE and integration are not currently supported with sctransform but will be soon.

As for covariate, MAST only supports numeric latent variable, so is not feasible for our analysis (categorical data). What's more, using covariate alone maybe not enough to tackle the noise in different samples (https://github.com/satijalab/seurat/issues/340). So in our analysis, we just omitted this and focused on the calculated DEGs that we are interested in. 

It would be nice to share your ideas If you have better ways to handle this.

**Comment by malisas** [comment]
Thanks for your response and clarification.

I am still learning about all this! So far, `scran::findMarkers(sce)` with a `block=sce$batch` argument seems straightforward. It is mentioned in [Orchestrating Single-Cell Analysis](https://osca.bioconductor.org/marker-detection.html#marker-batch). Chapter 14 of OSCA (Multi-sample comparisons) is also very informative. Certainly, it seems there are trade-offs for each approach, and I am still trying to decide what will work best for my own applications.

**Edit**: 

1. I just realized that `Seurat::FindConservedMarkers(seurat.obj)` with `grouping.var="batch"` is analogous to `scran::findMarkers(sce)` with a `block=sce$batch` argument (and it is actually mentioned in the issue you linked above). I think this is a kind of "meta analysis" because you find the markers for each batch independently and then combine the results. More info [here](https://satijalab.org/seurat/v3.1/immune_alignment.html). I wonder if this might be appropriate for your project. You wouldn't need to convert your data to be compatible with `scran`.

2. I also realized that I should have probably distinguished between "finding markers" and "differential expression analysis". People tend to use "finding markers" to mean the "identification of cluster-specific markers" (although it _is_ a kind of DE analysis). Whereas "Differential Expression" tends to refer more to finding differences across experimental conditions of interest. 
I'll change the title to clarify this.
